.bluemix/pipeline.yml

```diff
@@ -27,19 +27,46 @@ stages:
         exit 1
       fi
 
-      # Create the conversation service and name it after the chatbot
-      if ! cf service $CHATBOT_NAME >/dev/null; then
-        cf create-service conversation free "$CHATBOT_NAME"
-      fi
+      # List the conversation services, for example:
+      # $ cf services | grep conversation
+      # Chatbot-20170822164726835   conversation          free                create succeeded
+      # Conversation-hf             conversation          free                create succeeded
+      # NOTE: the -c option for grep counts number of rows
+      result=$(cf services | grep -c conversation)
 
-      # Also create similarly named service key / credential for the service
-      if ! cf service-key $CHATBOT_NAME $CHATBOT_NAME >/dev/null; then
+      # if no consersation exists, create service and service key, name after the chatbot
+      if [ "$result" -eq 0 ]; then
+        cf create-service conversation free "$CHATBOT_NAME"
         cf create-service-key "$CHATBOT_NAME" "$CHATBOT_NAME"
+      # if more than one exists, issue a warning
+      elif [ "$result" -gt 1 ]; then
+        echo "found more than one conversation service, using the first found"
+        # TODO: Use the paid service over the free?
       fi
+      # Finally, we get the service name, either newly created or first result
+      service_name=$(cf services | grep conversation | cut -d " " -f1)
```

Comment:
unfortunately, this won't work if $CHATBOT_NAME has a space... hmm...

Reply:
spoke with the CF folks in their #cli slack, and they didn't provide a better solution. though with mark's suggestion we are a better shape since the space was coming from the chatbot name, mostly. if the user creates the service via BAE then it won't have a space (it'll be BAE+datetime), and through the UI it doesn't have one by default. So though it's possible to create a service with a space, it's unlikely (famous last words)